### PR TITLE
feat(clients/golang): Added Golang client for create workflow instance and some test cases

### LINKED ISSUE #1095: I can create a workflow instance using the go client

### PR DESCRIPTION
**Pull request by jsam**
closes #1095

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 9bd899bf453b8d787970c1da58760d452c92cd36
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 8a4822dd765b297888ac78b1ea5e3a658111cbd1
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 7e1ab07ef53679e7c4bd4d2e049b4b22220069cb
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

File: clients/go/api.go [hunk 1]
```diff
@@ -7,6 +7,7 @@ import (
 type ZBClient interface {
 	NewHealthCheckCommand() *commands.HealthCheckCommand
 	NewDeployWorkflowCommand() *commands.DeployCommand
+    NewCreateInstanceCommand() commands.CreateInstanceCommandStep1
```
Comment (menski) [inline clients/go/api.go]: Can we auto format our go code? I saw multiple times different indention use inside the code base

File: clients/go/commands/createInstance_command.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    cmd.request.Version = version
+    return cmd
+}
+
+func (cmd *CreateInstanceCommand) LatestVersion() CreateInstanceCommandStep3 {
+    cmd.request.Version = LatestVersion
+    return cmd
+}
+
+func (cmd *CreateInstanceCommand) LatestVersionForce() CreateInstanceCommandStep3 {
```
Comment (menski) [inline clients/go/commands/createInstance_command.go]: This method is not supported anymore with the latest workflow deployment changes in the broker and was also removed from the java API with https://github.com/zeebe-io/zeebe/commit/edcfbb02743566cd4b7ede7643082237db247f0f#diff-1137e2ae25ebb2c456575400eb01b06cL76

I apparently forgot to delete the constant from the Java API which I will catch up on that later

File: clients/java/src/test/java/io/zeebe/client/ZeebeClientTest.java [hunk 3]
```diff
@@ -125,4 +125,26 @@ public void shouldNotFailIfClosedTwice() {
     client.close();
     client.close();
   }
+
+  public void shouldCreateWorkflowInstances() throws URISyntaxException {
```
Comment (menski) [inline clients/java/src/test/java/io/zeebe/client/ZeebeClientTest.java]: I think we don't need this that as we already have https://github.com/zeebe-io/zeebe/commit/edcfbb02743566cd4b7ede7643082237db247f0f#diff-db5f7e2bc4e08c03e2a5dbd5c829402fR90

File: gateway/src/test/java/io/zeebe/gateway/CreateWorkflowInstanceTest.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+import io.zeebe.gateway.util.RecordingStreamObserver;
+import io.zeebe.util.sched.future.ActorFuture;
+import io.zeebe.util.sched.future.CompletableActorFuture;
+import io.zeebe.util.sched.testing.ControlledActorSchedulerRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mock;
+
+public class CreateWorkflowInstanceTest {
```
Comment (menski) [inline gateway/src/test/java/io/zeebe/gateway/CreateWorkflowInstanceTest.java]: I personally still see no values in these test as they are superseded by the tests in the java client test suite without mocking. If it is okay for you I would remove them.

File: gateway/src/test/java/io/zeebe/gateway/ResponseMapperTest.java [hunk 5]
```diff
@@ -58,4 +60,16 @@ public void shouldTestDeployWorkflowMapping() {
               assertTrue(factory.contains(workflow));
             });
   }
+
+  @Test
+  public void shouldTestCreateWorkflowInstanceMapping() {
```
Comment (menski) [inline gateway/src/test/java/io/zeebe/gateway/ResponseMapperTest.java]: Personally I don't see value in these tests either as they are more boiler plate then what they add as value compared to the tests in the client test suite. Also you now had to change the implementation of the 3 entities just to make it easier to create the boiler plate for these tests. Not really liking it.

**Review by menski — CHANGES_REQUESTED** [review]
I would personally remove the java part from this PR and fix the go formatting. Then it's good to merge.

File: clients/go/commands/createInstance_command.go (hunk 2 above)
Comment (jsam) [reply clients/go/commands/createInstance_command.go]: Ok, I removed it.

File: clients/go/api.go (hunk 1 above)
Comment (jsam) [reply clients/go/api.go]: Strange, looks like Goland update forces new formatting defaults. I will gofmt entire project again. We should hook it up on git hook.

**Review by menski — APPROVED** [review]

**Pull request merged**